src/roman_datamodels/datamodels.py

```diff
@@ -312,10 +319,168 @@ class RampFitOutputModel(DataModel):
     pass
 
 
+class AssociationsModel(DataModel):
+    # Need an init to allow instantiation from a JSON file
+    pass
+
+
 class GuidewindowModel(DataModel):
     pass
 
 
+class ModelContainer(Sequence):
+    """
+    A container for holding DataModels.
+
+    This functions like a list for holding DataModel objects.  It can be
+    iterated through like a list, DataModels within the container can be
+    addressed by index, and the datamodels can be grouped into a list of
+    lists for grouped looping, useful for NIRCam where grouping together
+    all detectors of a given exposure is useful for some pipeline steps.
+
+    Parameters
+    ----------
+    init : file path, list of DataModels, or None
+
+        - file path: initialize from an association table
+
+        - list: a list of DataModels of any type
+
+        - None: initializes an empty `ModelContainer` instance, to which
+          DataModels can be added via the ``append()`` method.
+
+    iscopy : bool
+        Presume this model is a copy. Members will not be closed
+        when the model is closed/garbage-collected.
+
+    Examples
+    --------
+    >>> container = ModelContainer(['file1.asdf', 'file2.asdf', ..., 'fileN.asdf'])
+    >>> for model in container:
+    ...     print(model.meta.filename)
+
+
+    Notes
+    -----
+        The optional parameters ``save_open`` and ``return_open`` can be
+        provided to control how the `DataModel` are used by the
+        :py:class:`ModelContainer`. If ``save_open`` is set to `False`, each input
+        `DataModel` instance in ``init`` will be written out to disk and
+        closed, then only the filename for the `DataModel` will be used to
+        initialize the :py:class:`ModelContainer` object.
+        Subsequent access of each member will then open the `DataModel` file to
+        work with it. If ``return_open`` is also `False`, then the `DataModel`
+        will be closed when access to the `DataModel` is completed. The use of
+        these parameters can minimize the amount of memory used by this object
+        during processing, with these parameters being used
+        by :py:class:`~jwst.outlier_detection.OutlierDetectionStep`.
+
+        .. warning::
+            Input files will be updated in-place with new ``meta`` attribute
+            values when ASN table's members contain additional attributes.
+
+    """
+
+    def __init__(self, init=None, iscopy=False, **kwargs):
```

Comment:
Since there's no reason to use `kwargs` (it's not being passed on to any other function or method from inside this class), I explicitly listed the three parameters in the constructor and updated the code accordingly.